indy_common/auth_rules.py

```diff
@@ -0,0 +1,245 @@
+    """
+    'AND' strategy implemented. If there is several RoleDefs, then all of this must be presented in incomming constraints
+    """
+    def is_accepted(self, other_constraint):
+        compares = []
+        for r in other_constraint:
+            for s in self:
+                if r.role == s.role and s.sig_count <= r.sig_count:
+                    compares.append(True)
+        return len(self) == len(compares) and all(compares)
+
+
+"""
+
+Rule's classes
+
+"""
+
+
+class AbstractRule(metaclass=ABCMeta):
+    def __init__(self,
+                 description: str,
+                 txn_type: str,
+                 default_auth_constraint: AuthConstraint,
+                 rule_id='',
+                 field='*',
+                 old_value='*',
+                 new_value='*'):
+        self.description = description
+        self.txn_type = txn_type
+        self.rule_id = rule_id
+        self.default_auth_constraint = default_auth_constraint
+        self.field = field
+        # self.old_value = '*' if old_value == '*' else re.escape(old_value)
+        # self.new_value = '*' if new_value == '*' else re.escape(new_value)
+        self.old_value = old_value
+        self.new_value = new_value
+        self.allow_field = False
+        self.allow_old_value = False
+        self.allow_new_value = False
+        self.allow_for_all = False
+        self._create_rule_id()
+        self._preprocessing()
+
+    def _create_rule_id(self) -> str:
+        if not self.rule_id:
+            self.rule_id = compile_rule_id(self.txn_type,
+                                           self.field,
+                                           '*' if self.old_value == '*' else re.escape(self.old_value),
+                                           '*' if self.new_value == '*' else re.escape(self.new_value))
+
+    @abstractmethod
+    def _preprocessing(self):
+        raise NotImplementedError()
+
+
+class RuleAdd(AbstractRule):
+    def _preprocessing(self):
+        self.allow_old_value = True
+        if self.field == '*':
+            self.allow_field = True
+        if self.new_value == '*':
+            self.allow_new_value = True
+        if self.default_auth_constraint == []:
+            self.allow_for_all = True
+
+
+class RuleRemove(AbstractRule):
+    def _preprocessing(self):
+        self.allow_old_value = True
+        self.allow_new_value = True
+        if self.field == '*':
+            self.allow_field = True
+        if self.default_auth_constraint == []:
+            self.allow_for_all = True
+
+
+class RuleEdit(AbstractRule):
+    def _preprocessing(self):
+        if self.field == '*':
+            self.allow_field = True
+        if self.old_value == '*':
+            self.allow_old_value = True
+        if self.new_value == '*':
+            self.allow_new_value = True
+        if self.default_auth_constraint == []:
+            self.allow_for_all = True
+
+
+"""
+
+Policy's clases
+
+"""
+
```

Comment:
I would split the file into `auth_constraints`, `auth_rules`, `auth_policies`, `authorizer`. Please create a special package for this.